2217 Single Copy Orthologs
==========================

The list of 2217 comes from *L. major*, *T. cruzi* and *T. brucei*

***

Scripts
-------
These scripts will give you an idea of how many of the 2217 single copy orthologs are found in your assembly.

* gce_tcruzi: this is a "genome completion evaluation" using *T. cruzi* 2217 SCO
	* Usage: gce_tcruzi <path to assembly fasta folder> <Organism short name>
	* ```/usr/global/blp/bin/gce_tcruzi /data6/tol/tol/Trypanosoma/Trypanosoma_cruzi_Marinkellei_TCC_344/gDNA/assembly/fasta/ Tmar```
* gce_lmajor: this is a "genome completion evaluation" using L.major 2217 SCO
	
***

Reference sequences
-------------------
* /data2/tol/scripts/GenomeCompletionByReferenceList/genelist_files/TcruziCL_ESM_proteins.fa
* /data2/tol/scripts/GenomeCompletionByReferenceList/genelist_files/lmaj_proteins.fa


***

/usr/global/blp/bin/formatdb


perl /usr/global/blp/bin/asgard -w -i TcruziCL_ESM_proteins.fa -d ~/blast_database/TcG_genes.faa -p blastp -a 10 -B

* lower the default evalue (1e-6) to 1e-2 in order to pick up all the single copy orthologs
* ```perl /usr/global/blp/bin/asgard -w -i TcruziCL_ESM_proteins.fa -d ~/blast_database/TcG_genes.fasta -p tblastn -a 10 -B -e 1 &>run.log &```
* do cut -f 1 blastres.file | sort | uniq | wc -l to see how many of the 2217 SCO you got, then rem_dup.
* After you remove duplicates do cut -f 2 blastres.nodup | sort | uniq | wc -l 


```
/home/nsheth/Util/general/intersect.pl <file1> <column> <file2> <column> > outfile
```

example:
````perl /home/nsheth/Util/general/intersect.pl /data6/tol/tol/Trypanosoma/Trypanosoma_cruzi_Marinkellei_TCC_344/gDNA/assembly/fasta/Tmar_2.2_gap/Tmar_genes.gff 8 Tmar_blastres_tcruziSCO.nodup 1 > Tmar_SCO.gff````

* intersect.pl <file 1> <columns> <file 2> <columns>
* finds lines from file 1 that match on selected columns in file 2
* if no columns are specified, entire lines must match

***


Heterozygosity
--------------
**define variables**

* reads: mapping file (reads2assembly.bam)
* database: assembly file

**heterozygosity**

* ````/usr/global/blp/BEDTools-Version-2.16.1/bin/intersectBed -abam $reads -b $scogff > ${org}_SCO.reads.bam````
* ````samtools mpileup -BQ0 -d 1000000 -f $database ${org}_SCO.reads.bam > ${org}_SCO.reads.bam.mpileup````
* ````perl /home/vnkoparde/scripts/pileup2GenomicPositionTable.pl ${org}_SCO.reads.bam.mpileup ${org}_SCO.reads.bam.mpileup.30.table ````
* ````perl /home/vnkoparde/scripts/getHeterozygosityFromGenomicPositionTable.v3.pl $scogff ${org}_SCO.reads.bam.mpileup.30.table  |awk 'NR==1;NR>1{print $0|"sort -k2,2"}' > ${org}_SCO.reads.30.heterozygosity````
* ````mv HT_stats.txt ${org}_SCO.reads.30.HT_stats.txt````
* ````cat ${org}_SCO.reads.30.heterozygosity````

**Determine average coverage for single copy orthologs**
* grep "^Tmar" Tmar_SCO.reads.30.heterozygosity | awk '{sum+=$4} END {print sum/NR}'

